documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -0,0 +1,263 @@
+
+### Store changes
+
+#### Update the `searchBy` filter in the `search` store
+
+Currently, this filter is shaped as other filters: it is a list of objects with
+`code`, `name` and `checked` properties. This is a possible value now: `{
+searchBy: { code: "creator", name: "filters.searchBy.creator", checked: true }}.
+
+Since this filter is mutually exclusive (you cannot search by both the creator
+and the tag, for example), this shape is very difficult to update.
+
+It is better to set the `searchBy` filter to have one of the possible values:
+`{ searchBy: <null|creator|source|tag> }`.
+
+We should also create a new method, `setSearchBy`, in the `search` store, that
+would allow directly set the `searchBy` filter value.
+
+### Analytics changes
+
+I am not sure what events we want to add, if any. Should we add `<VIEW>_CLICKED`
+events for the new views, or would page views be sufficient?
+
+Should we add a new `SELECT_COLLECTION_ITEM` event similar to
+`SELECT_SEARCH_RESULT`?
+
+The clicks on external creator link in the `VCollectionHeader` should be tracked
+as `VISIT_CREATOR_LINK` events. We don't have a special event for visiting
+source, should we add one, or should they be tracked as generic
+`VISIT_EXTERNAL_LINK`?
+
+### Tests
+
+We should add visual-regression tests for the new views. To minimize flakiness
+due to slow loading of the images, we should probably use the
+`{ filter: brightness(0%); }` trick for the images on the page.
+
+The search store tests should be updated to reflect the changes to the filters.
+
+## Dependencies
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+These views potentially might cause more load on our infrastructure due to
+increase in scraping activity.
+
+### Tools & packages
+
+<!-- Describe any tools or packages which this work might be dependent on. If multiple options are available, try to list as many as are reasonable with your own recommendation. -->
+
+No new tools or packages are necessary.
+
+### Other projects or work
+
+<!-- Note any projects this plan is dependent on. -->
+
+## Design
+
+<!-- Note any design requirements for this plan. -->
+
+Figma designs in the dev mode:
+https://www.figma.com/file/niWnCgB7K0Y4e4mgxMrnRC/Additional-search-views?node-id=22%3A13656&mode=dev
+
+## Parallelizable streams
+
+<!-- What, if any, work within this plan can be parallelized? -->
+
+The work on components (`VCreatorLink`, `VSourceLink`, `VMediaTag`), and on the
+search store changes can be parallelized.
+
+We should create a separate branch for this work because adding the links to the
+collection pages on the single result pages (`VCreatorLink`, `VSourceLink`,
+`VMediaTag`) before the collection pages are ready is not possible.
```

Comment:
Should it be feature-flagged instead, Olga? I would think so. If you agree please update the plan to reflect that.

Reply:
Strongly agree on feature-flagging rather than feature-branching whenever possible.